host/allwinner/a20.go

```diff
@@ -0,0 +1,219 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// This file contains pin mapping information that is specific to the Allwinner
+// A20 model.
+
+package allwinner
+
+import (
+	"strings"
+
+	"periph.io/x/periph/conn/pin"
+	"periph.io/x/periph/host/sysfs"
+)
+
+// mappingA20 describes the mapping of the A20 processor gpios to their
+// alternate functions.
+//
+// It omits the in & out functions which are available on all gpio.
+//
+// The mapping comes from the datasheet page 241:
+// http://dl.linux-sunxi.org/A20/A20%20User%20Manual%202013-03-22.pdf
+var mappingA20 = map[string][5]pin.Func{
+	"PA0":  {"ERXD3", "SPI1_CS0", "UART2_RTS", "GRXD3"},
+	"PA1":  {"ERXD2", "SPI1_CLK", "UART2_CTS", "GRXD2"},
+	"PA2":  {"ERXD1", "SPI1_MOSI", "UART2_TX", "GRXD1"},
+	"PA3":  {"ERXD0", "SPI1_MISO", "UART2_RX", "GRXD0"},
+	"PA4":  {"ETXD3", "SPI1_CS1", "", "GTXD3"},
+	"PA5":  {"ETXD2", "SPI3_CS0", "", "GTXD2"},
+	"PA6":  {"ETXD1", "SPI3_CLK", "", "GTXD1"},
+	"PA7":  {"ETXD0", "SPI3_MOSI", "", "GTXD0"},
+	"PA8":  {"ERXCK", "SPI3_MISO", "", "CRXCK"},
+	"PA9":  {"ERXERR", "SPI3_CS1", "", "GNULL", "I2S1_MCLK"},
+	"PA10": {"ERXDV", "", "UART1_TX", "GRXCTL"},
+	"PA11": {"EMDC", "", "UART1_RX", "GMDC"},
+	"PA12": {"EMDIO", "UART6_TX", "UART1_RTS", "GMDIO"},
+	"PA13": {"ETXEN", "UART6_RX", "UART1_CTS", "GTXCTL"},
+	"PA14": {"ETXCK", "UART7_TX", "UART1_DTR", "GNULL", "I2S1_BCLK"},
+	"PA15": {"ECRS", "UART7_RX", "UART1_DSR", "GTXCK", "I2S1_LRCK"},
+	"PA16": {"ECOL", "CAN_TX", "UART1_DCD", "GCLKIN", "I2S1_DO"},
+	"PA17": {"ETXERR", "CAN_RX", "UART1_RING", "GNULL", "I2S1_DI"},
+	"PB0":  {"I2C0_SCK"},
```

Comment:
Please use SCL instead of SCK for I2C pins. Ref: https://periph.io/x/periph/conn/i2c#pkg-constants